### DISCUSSION TITLE
Relationship between ParticleTile::resize() and particles.tile_size ?

### DISCUSSION THREAD

**Discussion by n8xm** [AMReX-Codes/amrex#2592] [Q&A] [answered]
What is the relationship between the `resize()` method of `ParticleTile` and the [input parameter](https://amrex-codes.github.io/amrex/docs_html/Particle.html#inputs-parameters) `particles.tile_size` ? 

* I looked at the [implementation of ParticleTile::resize() in AMReX_ParticleTile.h](https://github.com/AMReX-Codes/amrex/blob/e7a725644a3eb72dba5c9cd74d35eff4d96af8a8/Src/Particle/AMReX_ParticleTile.H#L324),  and it appears to resize the `AoS` and `SoA` data structures associated with the ParticleTile. From my [earlier discussion](https://github.com/AMReX-Codes/amrex/discussions/2497) with @atmyers, it sounds like `particles.tile_size` controls the number of `ParticleTile` objects per grid, when `do_tiling` is set to `1`. 

* In addition, from what I can tell, `particles.tile_size` refers to a size in terms of *number of grid cells*, whereas `ParticleTile::resize()` changes the *number of particles* that the `ParticleTile` object owns.  

Am I correct, or am I missing something?

**Comment by atmyers** [comment, marked as answer]
Hi Nathan,

You are correct. `particles.tile_size` is in terms of number of grid cells and controls the number of `ParticleTile`s per grid. `ParticleTile::resize()` adjusts the lengths of the `AoS` and `SoA` data structures on a given `ParticleTile` in terms of the number of particles. A common use for `ParticleTile::resize()` would be to allocate space for some number of particles on a tile, then fill in the data for each particle.
